Components/Queues/PhpAmqpLibQueueManager.php

```diff
@@ -0,0 +1,309 @@
+
+    /**
+     * Maximum number of messages to consume if param killAfter exists
+     * @var int
+     */
+    public $max;
+
+    /**
+     * PhpAmqpLibQueueManager constructor.
+     * @param \AMQPConnection[] $connections
+     */
+    public function __construct($connections = null, int $max = -1, string $format = 'json', SerializableInterface $serializer = null)
+    {
+        $this->connections = $connections;
+        $this->shoudlStop = false;
+        $this->max = $max;
+        $this->format = $format;
+        $this->serializer = $serializer ?? SerializerBuilder::create()->build();
+    }
+
+    /**
+     * PhpAmqpLibQueueMessage destructor
+     * @throws \Exception
+     * @return
+     */
+    public function __destruct()
+    {
+        try {
+            if ($this->isConnected()) {
+                return $this->disconnect();
+            }
+        } catch (\Exception $exception) {
+            throw new \Exception($exception->getMessage());
+        }
+    }
+
+    /**
+     * Opens a connection with a queuing system.
+     * @param bool $shuffle Shuffle connections to avoid connecting to the seame endpoint everytime
+     * @throws \AMQPException
+     */
+    public function connect(bool $shuffle = true): void
+    {
+        if (!$this->isConnected()) {
+            if (empty($this->connections)) {
+                throw new \InvalidArgumentException('You have to specify at least one connection.');
+            }
+
+            if ($shuffle) {
+                shuffle($this->connections);
+            }
+
+            $this->connection = null;
+            $tested = [];
+
+            foreach ($this->connections as $connection) {
+                try {
+                    $connection->connect();
+                    $this->connection = $connection;
+                    break;
+                } catch (\AMQPConnectionException $e) {
+                    $tested[] = "{$connection->getHost()}: {$e->getMessage()}";
+                }
+            }
+
+            if (!$this->connection) {
+                throw new \RuntimeException(sprintf('Unable to connect to any of the following hosts:%s%s', PHP_EOL, implode(PHP_EOL, $tested)));
+            }
+        }
+
+        if (!$this->channel) {
+            $this->declareChannel();
+        }
+    }
+
+    /**
+     * Destroys the connection with the queuing system.
+     */
+    public function disconnect()
+    {
+        try {
+            if ($this->isConnected()) {
+                $this->connections[0]->close();
+            }
+            $this->connection = null;
+        } catch (\Exception $exception) {
+            throw new \Exception($exception->getMessage());
+        }
+    }
+
+    /**
+     * Returns true if a connection already exists with the queueing system, false otherwise.
+     * @return bool
+     */
+    public function isConnected(): bool
+    {
+        return $this->connections[0]  ? $this->connections[0]->isConnected() : false;
```

Comment:
Here is the same, we have the chance to have different connections and not always [0] will be the connection used

Reply:
This class was removed.